addons/account/models/account_move_line.py

```diff
@@ -3058,12 +3052,22 @@ def _reconciled_lines(self):
             ids.append(aml.id)
         return ids
 
-    def _all_reconciled_lines(self):
-        reconciled = self
+    def _reconciled_by_number(self):
         matching_numbers = [n for n in set(self.mapped('matching_number')) if n]
```

Comment:
Add a docstring. It's not obvious you are collecting move lines that are not in `self`